src/ray/core_worker/core_worker.h

```diff
@@ -1414,7 +1414,7 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
                       ObjectID object_id, void *py_future);
 
   /// Whether we are shutting down and not running further tasks.
```

Comment:
maybe some comments in the code about why it needs to be atomic? Right now our threading model is not very good, some comments about this will be useful for future reviewers.